adapters/copy.go

```diff
@@ -0,0 +1,29 @@
+package adapters
+
+import (
+	"github.com/smartcontractkit/chainlink/store"
+	"github.com/smartcontractkit/chainlink/store/models"
+	"github.com/tidwall/gjson"
+)
+
+// Copy obj keys refers to which value to copy inside `data`,
+// each obj value refers to where to copy the value to inside `data`
+type Copy struct {
+	models.JSON
+}
+
+// Perform returns the copied values from the desired mapping within the `data` JSON object
+func (c *Copy) Perform(input models.RunResult, store *store.Store) models.RunResult {
+	var err error
+	c.ForEach(func(from, to gjson.Result) bool {
+		val := input.Data.Get(from.String())
+		if val.Exists() && len(to.String()) > 0 {
+			input.Data, err = input.Data.Add(to.String(), val.String())
```

Comment:
You are assigning `err` multiple times as a result of the `ForEach` method looping over the keys. Should this be a `multierr.Append(...)`? Otherwise you will clobber an real erroneous state with a successful one:

```json
{
  "badkey": {"thiswill": "stillnoterror"},
  "goodkey": "ok"
}
```

That `err` scenario is pretty hard, if not impossible to hit so my example doesn't do it justice. But the issue still remains that a real error could get overwritten.

Reply:
I had it set-up with multierr when initially working on it, but ended up removing it because if you return false in the iterator, it breaks the loop. In my tidying up after, I removed the conditions where it would return false. I'll get multierr back in.

Yeah, I tried real hard to get it to error, after delving into the module it seemed the only thing which causes it to error is an invalid JSON object. With that, I can't really think of an achievable way to test the err condition.